### PR TITLE
feat: Block subscriptions if there are open invoices on the customer 

### PR DESCRIPTION
**Pull request by xrdt**
## Problem

~The billing changes must go in first: https://github.com/PostHog/billing/pull/583~ Ok billing changes are in.

There's a limitation in the [unsubscription flow](https://github.com/PostHog/billing/pull/573) where we may not be able to pay the invoice created for the billing cycle that just closed upon unsubscription. This PR closes that loop by preventing subscriptions and additions to a subscription (subscribing to additional products or add-ons if we detect open invoices on the stripe customer). We display a banner at the top of the /billing page (this is distinct from the billingAlerts as these _only_ exist on the /billing page) saying how many open invoices there are and link to the invoice, if possible. If the invoice link is not available, we link to the stripe portal, where users can look through a full list of invoices on their account and manually close them out. This message about not being able to subscribe is also displayed as a tooltip on disabled Subscribe / Add buttons on individual products. 

Note that we are by default considering open invoices with total amount due > $10.

Demo: 

https://github.com/PostHog/posthog/assets/21014901/1c2823b8-1fdb-444e-baf6-2eb50581c62d

## Changes

👉 _Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review._

### CODE REVIEW

File: ee/api/billing.py [hunk 1]
```diff
@@ -131,13 +131,52 @@ def deactivate(self, request: Request, *args: Any, **kwargs: Any) -> HttpRespons
...
+                        "code": detail_object.get("code"),
+                    },
                     status=status.HTTP_400_BAD_REQUEST,
                 )
             else:
                 raise e
         return self.list(request, *args, **kwargs)
 
+    @action(methods=["GET"], detail=False)
+    def get_open_invoices(self, request: Request, *args: Any, **kwargs: Any) -> HttpResponse:
```
Comment (zlwaterfield) [inline ee/api/billing.py:146]: Should this also be `get_invoices` and pass the status through?

Comment (xrdt) [reply ee/api/billing.py:146]: Yeah, that's a good point.

File: frontend/src/scenes/billing/billingLogic.tsx [hunk 2]
```diff
@@ -190,6 +201,41 @@ export const billingLogic = kea<billingLogicType>([
...
+                                    numOpenInvoices == 1 && jsonRes['link']
+                                        ? jsonRes['link']
+                                        : values.billing?.stripe_portal_url,
+                                children: 'View invoices',
+                                targetBlank: true,
+                            }
+                            return {
+                                status: 'warning',
+                                message: `You have ${numOpenInvoices} open invoice${
+                                    numOpenInvoices > 1 ? 's' : ''
```
Comment (xrdt) [inline frontend/src/scenes/billing/billingLogic.tsx:225]: Would appreciate a review on language here and overall flow @raquelmsmith

**Review by zlwaterfield — APPROVED** [review]

File: frontend/src/scenes/billing/billingLogic.tsx (hunk 2 above)
Comment (xrdt) [reply frontend/src/scenes/billing/billingLogic.tsx:225]: Going to merge for now and we can iterate.

**Pull request merged**